diff --git a/rcl/test/rcl/test_node.cpp b/rcl/test/rcl/test_node.cpp
index 20adac0..342fcdc 100644
--- a/rcl/test/rcl/test_node.cpp
+++ b/rcl/test/rcl/test_node.cpp
@@ -421,6 +421,7 @@ TEST_F(TestNodeFixture, test_rcl_node_life_cycle) {
   EXPECT_EQ(RCL_RET_OK, ret);
 }
 
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST_F(TestNodeFixture, test_rcl_node_init_with_internal_errors) {
   // We always call rcutils_logging_shutdown(), even if we didn't explicitly
   // initialize it.  That's because some internals of rcl may implicitly
@@ -1014,6 +1015,7 @@ TEST_F(TestNodeFixture, test_rcl_node_resolve_name) {
 
 /* Tests special case node_options
  */
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST_F(TestNodeFixture, test_rcl_get_disable_loaned_message) {
   {
     EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, rcl_get_disable_loaned_message(nullptr));