diff --git a/rcl/test/rcl/test_domain_id.cpp b/rcl/test/rcl/test_domain_id.cpp
index 6345e7d..a65adf2 100644
--- a/rcl/test/rcl/test_domain_id.cpp
+++ b/rcl/test/rcl/test_domain_id.cpp
@@ -54,6 +54,7 @@ TEST(TestGetDomainId, test_nominal) {
   rcl_reset_error();
 }
 
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST(TestGetDomainId, test_mock_get_default_domain_id) {
   auto mock = mocking_utils::patch_and_return(
     "lib:rcl", rcutils_get_env, "argument env_name is null");