diff --git a/rcl/test/rcl/test_service_event_publisher.cpp b/rcl/test/rcl/test_service_event_publisher.cpp
index e39a6c6..966e954 100644
--- a/rcl/test/rcl/test_service_event_publisher.cpp
+++ b/rcl/test/rcl/test_service_event_publisher.cpp
@@ -134,6 +134,7 @@ TEST_F(TestServiceEventPublisherFixture, test_service_event_publisher_nominal)
   EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
 }
 
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST_F(TestServiceEventPublisherFixture, test_service_event_publisher_init_and_fini)
 {
   rcl_service_event_publisher_t service_event_publisher =