diff --git a/rcl/test/rcl/test_rmw_impl_id_check_func.cpp b/rcl/test/rcl/test_rmw_impl_id_check_func.cpp
index 62c3f2e..2085c9c 100644
--- a/rcl/test/rcl/test_rmw_impl_id_check_func.cpp
+++ b/rcl/test/rcl/test_rmw_impl_id_check_func.cpp
@@ -82,7 +82,8 @@ MOCKING_UTILS_BOOL_OPERATOR_RETURNS_FALSE(rcutils_allocator_t, >)
 MOCKING_UTILS_BOOL_OPERATOR_RETURNS_FALSE(rcutils_allocator_t, !=)
 
 // Mock internal calls to external libraries to fail
-TEST(TestRmwCheck, test_mock_rmw_impl_check) {
+// TODO(asymingt) re-enable when we decide how to handle a RMW_IMPLEMENTATION environent variable.
+TEST(TestRmwCheck, DISABLED_test_mock_rmw_impl_check) {
   {
     // Fail reading RMW_IMPLEMENTATION_ENV_VAR_NAME
     auto mock = mocking_utils::patch_and_return(