diff --git a/rcl/test/rcl/test_logging.cpp b/rcl/test/rcl/test_logging.cpp
index dcad345..c1471ea 100644
--- a/rcl/test/rcl/test_logging.cpp
+++ b/rcl/test/rcl/test_logging.cpp
@@ -134,6 +134,7 @@ TEST(TestLogging, test_logging_rosout_enabled) {
   }
 }
 
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST(TestLogging, test_failing_external_logging_configure) {
   const char * ext_lib_flag = "--enable-" RCL_LOG_EXT_LIB_FLAG_SUFFIX;
   const char * argv[] = {"test_logging", RCL_ROS_ARGS_FLAG, ext_lib_flag};
@@ -168,6 +169,7 @@ TEST(TestLogging, test_failing_external_logging_configure) {
   }
 }
 
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST(TestLogging, test_failing_logger_level_configure) {
   const char * argv[] = {
     "test_logging", RCL_ROS_ARGS_FLAG,
@@ -193,6 +195,7 @@ TEST(TestLogging, test_failing_logger_level_configure) {
   }
 }
 
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST(TestLogging, test_failing_external_logging) {
   const char * stdout_flag = "--disable-" RCL_LOG_STDOUT_FLAG_SUFFIX;
   const char * ext_flag = "--enable-" RCL_LOG_EXT_LIB_FLAG_SUFFIX;