diff --git a/rcl/test/rcl/test_service.cpp b/rcl/test/rcl/test_service.cpp
index 0d637c7..0050aa9 100644
--- a/rcl/test/rcl/test_service.cpp
+++ b/rcl/test/rcl/test_service.cpp
@@ -462,6 +462,7 @@ MOCKING_UTILS_BOOL_OPERATOR_RETURNS_FALSE(rcutils_allocator_t, !=)
 
 /* Test failed service initialization using mocks
  */
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST_F(TestServiceFixture, test_fail_ini_mocked) {
   const rosidl_service_type_support_t * ts = ROSIDL_GET_SRV_TYPE_SUPPORT(
     test_msgs, srv, BasicTypes);
@@ -526,6 +527,7 @@ TEST_F(TestServiceFixture, test_fail_ini_mocked) {
 
 /* Test failed service finalization using mocks
  */
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST_F(TestServiceFixture, test_fail_fini_mocked) {
   const rosidl_service_type_support_t * ts = ROSIDL_GET_SRV_TYPE_SUPPORT(
     test_msgs, srv, BasicTypes);
@@ -550,6 +552,7 @@ TEST_F(TestServiceFixture, test_fail_fini_mocked) {
 
 /* Test failed service take_request_with_info using mocks and nullptrs
  */
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST_F(TestServiceFixture, test_fail_take_request_with_info) {
   const rosidl_service_type_support_t * ts = ROSIDL_GET_SRV_TYPE_SUPPORT(
     test_msgs, srv, BasicTypes);
@@ -618,6 +621,7 @@ TEST_F(TestServiceFixture, test_fail_take_request_with_info) {
 
 /* Test failed service send_response using mocks and nullptrs
  */
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST_F(TestServiceFixture, test_fail_send_response) {
   const rosidl_service_type_support_t * ts = ROSIDL_GET_SRV_TYPE_SUPPORT(
     test_msgs, srv, BasicTypes);