diff --git a/rcl/test/rcl/test_validate_enclave_name.cpp b/rcl/test/rcl/test_validate_enclave_name.cpp
index 862e244..7fadc20 100644
--- a/rcl/test/rcl/test_validate_enclave_name.cpp
+++ b/rcl/test/rcl/test_validate_enclave_name.cpp
@@ -28,6 +28,7 @@
 
 #include "../mocking_utils/patch.hpp"
 
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST(TestValidateEnclaveName, test_validate) {
   int validation_result;
   size_t invalid_index;
@@ -77,6 +78,7 @@ TEST(TestValidateEnclaveName, test_validate) {
   }
 }
 
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST(TestValidateEnclaveName, test_validate_on_internal_error) {
   int validation_result;
   size_t invalid_index;