diff --git a/rcl/test/rcl/test_wait.cpp b/rcl/test/rcl/test_wait.cpp
index 841a0fe..a262fae 100644
--- a/rcl/test/rcl/test_wait.cpp
+++ b/rcl/test/rcl/test_wait.cpp
@@ -843,6 +843,7 @@ TEST_F(WaitSetTestFixture, wait_set_get_allocator) {
 }
 
 // Test wait set init failure cases using mocks
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST_F(WaitSetTestFixture, wait_set_failed_init) {
   rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
   // Mock rmw implementation to fail init
@@ -855,6 +856,7 @@ TEST_F(WaitSetTestFixture, wait_set_failed_init) {
 }
 
 // Test wait set fini failure cases using mocks
+// TODO(asymingt) enable once mocking works in Bazel.
 TEST_F(WaitSetTestFixture, wait_set_failed_fini) {
   rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
   rcl_ret_t ret =