diff --git a/rcl_yaml_param_parser/test/test_parser_multiple_nodes.cpp b/rcl_yaml_param_parser/test/test_parser_multiple_nodes.cpp
index 5458bfa..ac7edf6 100644
--- a/rcl_yaml_param_parser/test/test_parser_multiple_nodes.cpp
+++ b/rcl_yaml_param_parser/test/test_parser_multiple_nodes.cpp
@@ -26,12 +26,13 @@
 #include "rcutils/testing/fault_injection.h"
 
 #include "./mocking_utils/patch.hpp"
+#include "./locate_test_assets.h"
 
 static char cur_dir[1024];
 
 TEST(RclYamlParamParserMultipleNodes, multiple_number_nodes) {
   rcutils_reset_error();
-  EXPECT_TRUE(rcutils_get_cwd(cur_dir, 1024)) << rcutils_get_error_string().str;
+  EXPECT_TRUE(get_test_asset_dir(cur_dir, 1024)) << rcutils_get_error_string().str;
   rcutils_allocator_t allocator = rcutils_get_default_allocator();
   char * test_path = rcutils_join_path(cur_dir, "test", allocator);
   ASSERT_TRUE(NULL != test_path) << rcutils_get_error_string().str;
@@ -69,7 +70,7 @@ TEST(RclYamlParamParserMultipleNodes, multiple_number_nodes) {
 TEST(RclYamlParamParserMultipleNodes, test_multiple_nodes_with_bad_allocator) {
   char cur_dir[1024];
   rcutils_reset_error();
-  EXPECT_TRUE(rcutils_get_cwd(cur_dir, 1024)) << rcutils_get_error_string().str;
+  EXPECT_TRUE(get_test_asset_dir(cur_dir, 1024)) << rcutils_get_error_string().str;
   rcutils_allocator_t allocator = rcutils_get_default_allocator();
   char * test_path = rcutils_join_path(cur_dir, "test", allocator);
   ASSERT_TRUE(NULL != test_path) << rcutils_get_error_string().str;