diff --git a/rcl_yaml_param_parser/test/test_parser.cpp b/rcl_yaml_param_parser/test/test_parser.cpp
index 1dec47d..5c61975 100644
--- a/rcl_yaml_param_parser/test/test_parser.cpp
+++ b/rcl_yaml_param_parser/test/test_parser.cpp
@@ -32,6 +32,7 @@
 
 #include "./mocking_utils/patch.hpp"
 #include "./time_bomb_allocator_testing_utils.h"
+#include "./locate_test_assets.h"
 
 TEST(RclYamlParamParser, node_init_fini) {
   rcutils_allocator_t allocator = rcutils_get_default_allocator();
@@ -400,7 +401,7 @@ TEST(RclYamlParamParser, test_yaml_node_struct_print) {
 TEST(RclYamlParamParser, test_parse_file_with_bad_allocator) {
   char cur_dir[1024];
   rcutils_reset_error();
-  EXPECT_TRUE(rcutils_get_cwd(cur_dir, 1024)) << rcutils_get_error_string().str;
+  EXPECT_TRUE(get_test_asset_dir(cur_dir, 1024)) << rcutils_get_error_string().str;
   rcutils_allocator_t allocator = rcutils_get_default_allocator();
   char * test_path = rcutils_join_path(cur_dir, "test", allocator);
   ASSERT_TRUE(NULL != test_path) << rcutils_get_error_string().str;
@@ -462,7 +463,7 @@ TEST(RclYamlParamParser, test_parse_file_with_bad_allocator) {
 TEST(RclYamlParamParser, test_parse_yaml_initialize_mock) {
   char cur_dir[1024];
   rcutils_reset_error();
-  EXPECT_TRUE(rcutils_get_cwd(cur_dir, 1024)) << rcutils_get_error_string().str;
+  EXPECT_TRUE(get_test_asset_dir(cur_dir, 1024)) << rcutils_get_error_string().str;
 
   rcutils_allocator_t allocator = rcutils_get_default_allocator();
   char * test_path = rcutils_join_path(cur_dir, "test", allocator);