diff --git a/rcl_yaml_param_parser/test/test_parser_multiple_params.cpp b/rcl_yaml_param_parser/test/test_parser_multiple_params.cpp
index 244a2cc..bd0bb35 100644
--- a/rcl_yaml_param_parser/test/test_parser_multiple_params.cpp
+++ b/rcl_yaml_param_parser/test/test_parser_multiple_params.cpp
@@ -26,12 +26,13 @@
 #include "rcutils/testing/fault_injection.h"
 
 #include "./mocking_utils/patch.hpp"
+#include "./locate_test_assets.h"
 
 static char cur_dir[1024];
 
 TEST(RclYamlParamParserMultipleParams, multiple_params) {
   rcutils_reset_error();
-  EXPECT_TRUE(rcutils_get_cwd(cur_dir, 1024)) << rcutils_get_error_string().str;
+  EXPECT_TRUE(get_test_asset_dir(cur_dir, 1024)) << rcutils_get_error_string().str;
   rcutils_allocator_t allocator = rcutils_get_default_allocator();
   char * test_path = rcutils_join_path(cur_dir, "test", allocator);
   ASSERT_TRUE(NULL != test_path) << rcutils_get_error_string().str;
@@ -67,7 +68,7 @@ TEST(RclYamlParamParserMultipleParams, multiple_params) {
 TEST(RclYamlParamParserMultipleParams, test_multiple_params_with_bad_allocator) {
   char cur_dir[1024];
   rcutils_reset_error();
-  EXPECT_TRUE(rcutils_get_cwd(cur_dir, 1024)) << rcutils_get_error_string().str;
+  EXPECT_TRUE(get_test_asset_dir(cur_dir, 1024)) << rcutils_get_error_string().str;
   rcutils_allocator_t allocator = rcutils_get_default_allocator();
   char * test_path = rcutils_join_path(cur_dir, "test", allocator);
   ASSERT_TRUE(NULL != test_path) << rcutils_get_error_string().str;