diff --git a/test/test_time.cpp b/test/test_time.cpp
index 5976cab..81b4655 100644
--- a/test/test_time.cpp
+++ b/test/test_time.cpp
@@ -16,6 +16,7 @@
 
 #include <chrono>
 #include <cinttypes>
+#include <iomanip>
 #include <thread>
 
 #include "osrf_testing_tools_cpp/memory_tools/memory_tools.hpp"
@@ -193,7 +194,8 @@ TEST_F(TestTimeFixture, test_rcutils_steady_time_now) {
 
 #if !defined(_WIN32)
 
-TEST_F(TestTimeFixture, test_rcutils_with_bad_system_clocks) {
+// TODO(asymingt): re-enable when mocking can be done on upstream symbols.
+TEST_F(TestTimeFixture, DISABLED_test_rcutils_with_bad_system_clocks) {
   {
     auto mock = mocking_utils::patch(
       "lib:rcutils", clock_gettime,