diff --git a/rosidl_generator_py/resource/_action.py.em b/rosidl_generator_py/resource/_action.py.em
index 77fbf87..0dd3891 100644
--- a/rosidl_generator_py/resource/_action.py.em
+++ b/rosidl_generator_py/resource/_action.py.em
@@ -4,31 +4,38 @@ from rosidl_pycommon import convert_camel_case_to_lower_case_underscore
 
 action_name = '_' + convert_camel_case_to_lower_case_underscore(action.namespaced_type.name)
 module_name = '_' + convert_camel_case_to_lower_case_underscore(interface_path.stem)
+import_suffix = 'action_' + action_name
 
 TEMPLATE(
     '_msg.py.em',
     package_name=package_name, interface_path=interface_path,
-    message=action.goal, import_statements=import_statements)
+    message=action.goal, import_statements=import_statements,
+    import_ifcode=action_name, import_suffix=import_suffix)
 TEMPLATE(
     '_msg.py.em',
     package_name=package_name, interface_path=interface_path,
-    message=action.result, import_statements=import_statements)
+    message=action.result, import_statements=import_statements,
+    import_ifcode=action_name, import_suffix=import_suffix)
 TEMPLATE(
     '_msg.py.em',
     package_name=package_name, interface_path=interface_path,
-    message=action.feedback, import_statements=import_statements)
+    message=action.feedback, import_statements=import_statements,
+    import_ifcode=action_name, import_suffix=import_suffix)
 TEMPLATE(
     '_srv.py.em',
     package_name=package_name, interface_path=interface_path,
-    service=action.send_goal_service, import_statements=import_statements)
+    service=action.send_goal_service, import_statements=import_statements,
+    import_ifcode=action_name, import_suffix=import_suffix)
 TEMPLATE(
     '_srv.py.em',
     package_name=package_name, interface_path=interface_path,
-    service=action.get_result_service, import_statements=import_statements)
+    service=action.get_result_service, import_statements=import_statements,
+    import_ifcode=action_name, import_suffix=import_suffix)
 TEMPLATE(
     '_msg.py.em',
     package_name=package_name, interface_path=interface_path,
-    message=action.feedback_message, import_statements=import_statements)
+    message=action.feedback_message, import_statements=import_statements,
+    import_ifcode=action_name, import_suffix=import_suffix)
 }@
 
 
@@ -41,7 +48,7 @@ class Metaclass_@(action.namespaced_type.name)(type):
     def __import_type_support__(cls):
         try:
             from rosidl_generator_py import import_type_support
-            module = import_type_support('@(package_name)')
+            module = import_type_support('@(package_name)__action_@(action_name)_s__rosidl_typesupport_c')
         except ImportError:
             import logging
             import traceback