diff --git a/test/test_filesystem_helper_std.cpp b/test/test_filesystem_helper_std.cpp
index 7f9beed..fe909b4 100644
--- a/test/test_filesystem_helper_std.cpp
+++ b/test/test_filesystem_helper_std.cpp
@@ -392,7 +392,11 @@ TEST(TestFilesystemHelper, remove_extension_no_extension)
 
 TEST(TestFilesystemHelper, get_cwd)
 {
-  std::string expected_dir = rcpputils::get_env_var("EXPECTED_WORKING_DIRECTORY");
+  // The current path in bazel is the test workspace, which is relative to
+  // the runfiles directory. Update exected path accordingly.
+  std::string runfiles_dir = rcpputils::get_env_var("RUNFILES_DIR");
+  std::string test_workspace = rcpputils::get_env_var("TEST_WORKSPACE");
+  std::string expected_dir = runfiles_dir + "/" + test_workspace;
   auto p = std::filesystem::current_path();
   EXPECT_EQ(expected_dir, p.string());
 }