diff --git a/rmw_zenoh_cpp/src/detail/rmw_context_impl_s.cpp b/rmw_zenoh_cpp/src/detail/rmw_context_impl_s.cpp
index 3e3d0b7..f144030 100644
--- a/rmw_zenoh_cpp/src/detail/rmw_context_impl_s.cpp
+++ b/rmw_zenoh_cpp/src/detail/rmw_context_impl_s.cpp
@@ -108,7 +108,7 @@ public:
           RMW_ZENOH_LOG_WARN_NAMED(
             "rmw_zenoh_cpp",
             "Unable to connect to a Zenoh router. "
-            "Have you started a router with `ros2 run rmw_zenoh_cpp rmw_zenohd`?");
+            "Have you started a router with `bazel run @rmw_zenoh_cpp//:rmw_zenohd`?");
         }
         if (++connection_attempts >= configured_connection_attempts.value()) {
           RMW_ZENOH_LOG_WARN_NAMED(