diff --git a/rclcpp_action/test/test_server_goal_handle.cpp b/rclcpp_action/test/test_server_goal_handle.cpp
index eb392c8b..874162b3 100644
--- a/rclcpp_action/test/test_server_goal_handle.cpp
+++ b/rclcpp_action/test/test_server_goal_handle.cpp
@@ -146,7 +146,6 @@ TEST_F(TestServerGoalHandle, succeed) {
   EXPECT_THROW(handle_->succeed(result), rclcpp::exceptions::RCLError);
 }
 
-
 TEST_F(TestServerGoalHandle, execute) {
   handle_->execute();
   EXPECT_FALSE(handle_->is_canceling());