diff --git a/rosidl_typesupport_protobuf_c/resource/srv__type_support.cpp.em b/rosidl_typesupport_protobuf_c/resource/srv__type_support.cpp.em
index 9e7e50a..701cfcc 100644
--- a/rosidl_typesupport_protobuf_c/resource/srv__type_support.cpp.em
+++ b/rosidl_typesupport_protobuf_c/resource/srv__type_support.cpp.em
@@ -19,7 +19,7 @@
 }@
 @# Included from rosidl_typesupport_protobuf_c/resource/idl__type_support.cpp.em
 @{
-from rosidl_cmake import convert_camel_case_to_lower_case_underscore
+from rosidl_pycommon import convert_camel_case_to_lower_case_underscore
 from rosidl_typesupport_protobuf import ros_service_name, ros_service_namespace, ros_service_type
 
 system_header_files = []