diff --git a/rclcpp/test/rclcpp/wait_set_policies/test_storage_policy_common.cpp b/rclcpp/test/rclcpp/wait_set_policies/test_storage_policy_common.cpp
index 8de89017..833ae82b 100644
--- a/rclcpp/test/rclcpp/wait_set_policies/test_storage_policy_common.cpp
+++ b/rclcpp/test/rclcpp/wait_set_policies/test_storage_policy_common.cpp
@@ -127,6 +127,7 @@ TEST_F(TestStoragePolicyCommon, rcl_wait_set_add_subscription_error) {
     wait_set.wait(),
     rclcpp::exceptions::RCLError);
 }
+
 TEST_F(TestStoragePolicyCommon, rcl_wait_set_add_guard_condition_error) {
   rclcpp::WaitSet wait_set;
   auto guard_condition = std::make_shared<rclcpp::GuardCondition>();