diff --git a/rclcpp/test/rclcpp/node_interfaces/test_node_graph.cpp b/rclcpp/test/rclcpp/node_interfaces/test_node_graph.cpp
index a7f361b4..3a5cca2b 100644
--- a/rclcpp/test/rclcpp/node_interfaces/test_node_graph.cpp
+++ b/rclcpp/test/rclcpp/node_interfaces/test_node_graph.cpp
@@ -290,7 +290,6 @@ TEST_F(TestNodeGraph, get_service_names_and_types_by_node_rcl_errors)
       " service names and types, leaking memory: error not set"));
 }
 
-
 TEST_F(TestNodeGraph, get_client_names_and_types_by_node_rcl_errors)
 {
   auto client = node()->create_client<test_msgs::srv::Empty>("node1_service");