diff --git a/rclc_lifecycle/src/rclc_lifecycle/rclc_lifecycle.c b/rclc_lifecycle/src/rclc_lifecycle/rclc_lifecycle.c
index 8bd9aac..1eaac56 100644
--- a/rclc_lifecycle/src/rclc_lifecycle/rclc_lifecycle.c
+++ b/rclc_lifecycle/src/rclc_lifecycle/rclc_lifecycle.c
@@ -34,6 +34,24 @@
 #include <lifecycle_msgs/srv/get_state.h>
 
 
+rclc_lifecycle_node_t
+rclc_get_zero_initialized_lifecycle_node(void)
+{
+  static rclc_lifecycle_node_t null_lifecycle_node;
+  // Setting all fields to 0
+  null_lifecycle_node.node = NULL;
+  null_lifecycle_node.state_machine = NULL;
+  memset(&null_lifecycle_node.callbacks, 0, sizeof(rclc_lifecycle_callback_map_t));
+  null_lifecycle_node.publish_transitions = false;
+  memset(&null_lifecycle_node.cs_req, 0, sizeof(lifecycle_msgs__srv__ChangeState_Request));
+  memset(&null_lifecycle_node.cs_res, 0, sizeof(lifecycle_msgs__srv__ChangeState_Response));
+  memset(&null_lifecycle_node.gs_req, 0, sizeof(lifecycle_msgs__srv__GetState_Request));
+  memset(&null_lifecycle_node.gs_res, 0, sizeof(lifecycle_msgs__srv__GetState_Response));
+  memset(&null_lifecycle_node.gas_req, 0, sizeof(lifecycle_msgs__srv__GetAvailableStates_Request));
+  memset(&null_lifecycle_node.gas_res, 0, sizeof(lifecycle_msgs__srv__GetAvailableStates_Response));
+  return null_lifecycle_node;
+}
+
 rcl_ret_t
 rclc_make_node_a_lifecycle_node(
   rclc_lifecycle_node_t * lifecycle_node,