diff --git a/rosidl_typesupport_cpp/rosidl_typesupport_cpp/__init__.py b/rosidl_typesupport_cpp/rosidl_typesupport_cpp/__init__.py
index f21ad1d..27c94b3 100644
--- a/rosidl_typesupport_cpp/rosidl_typesupport_cpp/__init__.py
+++ b/rosidl_typesupport_cpp/rosidl_typesupport_cpp/__init__.py
@@ -23,7 +23,7 @@ def generate_cpp(generator_arguments_file, type_supports):
     :param type_support: List of type supports to be used
     """
     mapping = {
-        'idl__type_support.cpp.em': '%s__type_support.cpp',
+        'idl__type_support.cpp.em': 'detail/%s__rosidl_typesupport_cpp.cpp',
     }
     return generate_files(
         generator_arguments_file, mapping,