diff --git a/rosidl_typesupport_cpp/resource/srv__type_support.cpp.em b/rosidl_typesupport_cpp/resource/srv__type_support.cpp.em
index e889773..d8e784e 100644
--- a/rosidl_typesupport_cpp/resource/srv__type_support.cpp.em
+++ b/rosidl_typesupport_cpp/resource/srv__type_support.cpp.em
@@ -36,6 +36,9 @@ include_parts = [package_name] + list(interface_path.parents[0].parts) + [
     'detail', convert_camel_case_to_lower_case_underscore(interface_path.stem)]
 include_base = '/'.join(include_parts)
 
+interface_type = "__".join(interface_path.parents[0].parts)
+interface_code = convert_camel_case_to_lower_case_underscore(interface_path.stem)
+
 header_files = [
     'cstddef',
     'rosidl_runtime_c/service_type_support_struct.h',
@@ -117,7 +120,7 @@ static _@(service.namespaced_type.name)_type_support_data_t _@(service.namespace
 
 static const type_support_map_t _@(service.namespaced_type.name)_service_typesupport_map = {
   @(len(type_supports)),
-  "@(package_name)",
+  "@(package_name)__@(interface_type)__@(interface_code)",
   &_@(service.namespaced_type.name)_service_typesupport_ids.typesupport_identifier[0],
   &_@(service.namespaced_type.name)_service_typesupport_symbol_names.symbol_name[0],
   &_@(service.namespaced_type.name)_service_typesupport_data.data[0],