diff --git a/rosidl_adapter/scripts/action2idl.py b/rosidl_adapter/scripts/action2idl.py
index 8385647..a7c3aa8 100644
--- a/rosidl_adapter/scripts/action2idl.py
+++ b/rosidl_adapter/scripts/action2idl.py
@@ -14,9 +14,27 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import argparse
+import pathlib
+
 from rosidl_adapter.action import convert_action_to_idl
 from rosidl_adapter.cli import convert_files_to_idl
 
 
 if __name__ == '__main__':
-    convert_files_to_idl('.action', convert_action_to_idl)
+    parser = argparse.ArgumentParser(prog='action2idl',
+        description='Transforms .srv files to .idl files')
+    parser.add_argument('-p', '--package-path', required=False, type=pathlib.Path)
+    parser.add_argument('-n', '--package-name', required=False, type=str)
+    parser.add_argument('input_file', type=pathlib.Path)
+    parser.add_argument('output_dir', type=pathlib.Path)
+    args = parser.parse_args()
+    if args.package_path is not None and args.package_name is not None:
+        convert_action_to_idl(
+            package_dir = pathlib.Path.cwd() / args.package_path,
+            package_name = args.package_name,
+            input_file = args.input_file,
+            output_dir = pathlib.Path.cwd() / args.output_dir
+        )
+    else:
+        convert_files_to_idl('.action', convert_action_to_idl)