diff --git a/rosidl_typesupport_protobuf_cpp/CMakeLists.txt b/rosidl_typesupport_protobuf_cpp/CMakeLists.txt
index 2413e03..05bb037 100644
--- a/rosidl_typesupport_protobuf_cpp/CMakeLists.txt
+++ b/rosidl_typesupport_protobuf_cpp/CMakeLists.txt
@@ -66,19 +66,18 @@ ament_python_install_package(${PROJECT_NAME})
 add_library(${PROJECT_NAME} SHARED
   src/identifier.cpp
   src/wstring_conversion.cpp)
-
 if(WIN32)
   target_compile_definitions(${PROJECT_NAME}
     PRIVATE "ROSIDL_TYPESUPPORT_PROTOBUF_CPP_BUILDING_DLL")
 endif()
-
 target_include_directories(${PROJECT_NAME} PUBLIC
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
-
-ament_target_dependencies(${PROJECT_NAME} rosidl_typesupport_protobuf rosidl_runtime_cpp)
-
-ament_export_dependencies(rosidl_runtime_cpp)
+  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
+  ${rosidl_typesupport_protobuf_INCLUDE_DIRS}
+)
+target_link_libraries(${PROJECT_NAME} PUBLIC
+  rosidl_runtime_cpp::rosidl_runtime_cpp
+)
 
 # Export old-style CMake variables
 ament_export_include_directories("include/${PROJECT_NAME}")