diff --git a/rosidl_typesupport_protobuf_cpp/bin/rosidl_typesupport_protobuf_cpp b/rosidl_typesupport_protobuf_cpp/bin/rosidl_typesupport_protobuf_cpp
index c8f50a5..33a8949 100644
--- a/rosidl_typesupport_protobuf_cpp/bin/rosidl_typesupport_protobuf_cpp
+++ b/rosidl_typesupport_protobuf_cpp/bin/rosidl_typesupport_protobuf_cpp
@@ -24,7 +24,7 @@ import pathlib
 import os
 import rosidl_typesupport_protobuf
 
-from rosidl_cmake import read_generator_arguments
+from rosidl_pycommon import read_generator_arguments
 from rosidl_typesupport_protobuf_cpp import generate_cpp
 
 def main(argv=sys.argv[1:]):