diff --git a/rosidl_typesupport_protobuf_cpp/cmake/rosidl_typesupport_protobuf_cpp_generate_interfaces.cmake b/rosidl_typesupport_protobuf_cpp/cmake/rosidl_typesupport_protobuf_cpp_generate_interfaces.cmake
index 88c8853..8ac37a2 100644
--- a/rosidl_typesupport_protobuf_cpp/cmake/rosidl_typesupport_protobuf_cpp_generate_interfaces.cmake
+++ b/rosidl_typesupport_protobuf_cpp/cmake/rosidl_typesupport_protobuf_cpp_generate_interfaces.cmake
@@ -148,25 +148,27 @@ set_target_properties(${rosidl_generate_interfaces_TARGET}${_target_suffix}
     "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_c>"
     "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/rosidl_typesupport_protobuf_cpp>"
     "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
+    "${rosidl_typesupport_interface_INCLUDE_DIRS}"
   )
 
 
-ament_target_dependencies(${rosidl_generate_interfaces_TARGET}${_target_suffix}
-  rmw
-  rosidl_typesupport_protobuf
-  rosidl_typesupport_protobuf_cpp
-  rosidl_typesupport_interface
+target_link_libraries(${rosidl_generate_interfaces_TARGET}${_target_suffix} PUBLIC
+  rmw::rmw
+  rosidl_typesupport_interface::rosidl_typesupport_interface
+  rosidl_typesupport_protobuf_cpp::rosidl_typesupport_protobuf_cpp
 )
 
 # Depend on dependencies
 foreach(_pkg_name ${rosidl_generate_interfaces_DEPENDENCY_PACKAGE_NAMES})
-  ament_target_dependencies(${rosidl_generate_interfaces_TARGET}${_target_suffix}
-    ${_pkg_name})
-  target_link_libraries(${rosidl_generate_interfaces_TARGET}${_target_suffix}
+  target_include_directories(${rosidl_generate_interfaces_TARGET}${_target_suffix} PUBLIC
+    ${${_pkg_name}_INCLUDE_DIRS})
+  target_link_libraries(${rosidl_generate_interfaces_TARGET}${_target_suffix} PUBLIC
     ${${_pkg_name}_LIBRARIES${_target_suffix}})
 endforeach()
 
-target_link_libraries(${rosidl_generate_interfaces_TARGET}${_target_suffix} ${Protobuf_LIBRARY})
+target_link_libraries(${rosidl_generate_interfaces_TARGET}${_target_suffix} PUBLIC
+  ${Protobuf_LIBRARY}
+)
 
 # Make top level generation target depend on this library
 add_dependencies(