diff --git a/rclpy/rclpy/impl/implementation_singleton.py b/rclpy/rclpy/impl/implementation_singleton.py
index 3e31228..b28bd56 100644
--- a/rclpy/rclpy/impl/implementation_singleton.py
+++ b/rclpy/rclpy/impl/implementation_singleton.py
@@ -26,8 +26,16 @@ For example, you might use it like this:
         # ...
 """
 
+import sys
+import os
+
 from rpyutils import import_c_library
 
 package = 'rclpy'
 
-rclpy_implementation = import_c_library('._rclpy_pybind11', package)
+_original_dlopen_flags = sys.getdlopenflags()
+try:
+    sys.setdlopenflags(os.RTLD_GLOBAL | os.RTLD_LAZY)
+    rclpy_implementation = import_c_library('._rclpy_pybind11', package)
+finally:
+    sys.setdlopenflags(_original_dlopen_flags)