diff --git a/rcl_logging_spdlog/test/test_logging_interface.cpp b/rcl_logging_spdlog/test/test_logging_interface.cpp
index 665a710..23ea02d 100644
--- a/rcl_logging_spdlog/test/test_logging_interface.cpp
+++ b/rcl_logging_spdlog/test/test_logging_interface.cpp
@@ -198,7 +198,9 @@ TEST_F(AllocatorTest, init_invalid)
   rcutils_reset_error();
 }
 
-TEST_F(AllocatorTest, init_failure)
+// In Bazel we will always have a valid path because TEST_UNDECLARED_OUTPUTS_DIR is always set
+// in a test context. So, this test cannot pass unless we are able to disable this.
+TEST_F(AllocatorTest, DISABLED_init_failure)
 {
   RestoreEnvVar home_var("HOME");
   RestoreEnvVar userprofile_var("USERPROFILE");