diff --git a/rosidl_typesupport_protobuf/include/rosidl_typesupport_protobuf/proto_descriptor_helper.hpp b/rosidl_typesupport_protobuf/include/rosidl_typesupport_protobuf/proto_descriptor_helper.hpp
index 95a71a2..db4d770 100644
--- a/rosidl_typesupport_protobuf/include/rosidl_typesupport_protobuf/proto_descriptor_helper.hpp
+++ b/rosidl_typesupport_protobuf/include/rosidl_typesupport_protobuf/proto_descriptor_helper.hpp
@@ -33,7 +33,7 @@
 #endif
 
 
-inline bool HasFile(const google::protobuf::FileDescriptorSet & fset_, const std::string & fname_)
+inline bool HasFile(const google::protobuf::FileDescriptorSet & fset_, const std::basic_string_view<char> fname_)
 {
   for (auto findex = 0; findex < fset_.file_size(); ++findex) {
     if (fset_.file(findex).name() == fname_) {