diff --git a/rclc_examples/src/example_lifecycle_node.c b/rclc_examples/src/example_lifecycle_node.c
index abb6cab..f16c502 100644
--- a/rclc_examples/src/example_lifecycle_node.c
+++ b/rclc_examples/src/example_lifecycle_node.c
@@ -83,7 +83,7 @@ int main(int argc, const char * argv[])
   // make it a lifecycle node
   printf("creating lifecycle node...\n");
   rcl_lifecycle_state_machine_t state_machine_ = rcl_lifecycle_get_zero_initialized_state_machine();
-  rclc_lifecycle_node_t lifecycle_node;
+  rclc_lifecycle_node_t lifecycle_node = rclc_get_zero_initialized_lifecycle_node();
   rc = rclc_make_node_a_lifecycle_node(
     &lifecycle_node,
     &my_node,