diff --git a/rosidl_adapter_proto/resource/srv.proto.em b/rosidl_adapter_proto/resource/srv.proto.em
index e25b279..a390055 100644
--- a/rosidl_adapter_proto/resource/srv.proto.em
+++ b/rosidl_adapter_proto/resource/srv.proto.em
@@ -18,7 +18,7 @@
 # ================================= Apache 2.0 =================================
 }@
 @{
-from rosidl_cmake import convert_camel_case_to_lower_case_underscore
+from rosidl_pycommon import convert_camel_case_to_lower_case_underscore
 import rosidl_parser.definition as rosidl
 
 from rosidl_adapter_proto import PROTO_PACKAGE_POSTFIX