diff --git a/rosidl_typesupport_c/resource/msg__type_support.cpp.em b/rosidl_typesupport_c/resource/msg__type_support.cpp.em
index b1340d9..3a1ed7f 100644
--- a/rosidl_typesupport_c/resource/msg__type_support.cpp.em
+++ b/rosidl_typesupport_c/resource/msg__type_support.cpp.em
@@ -9,8 +9,11 @@ include_parts = [package_name] + list(interface_path.parents[0].parts) + [
     'detail', convert_camel_case_to_lower_case_underscore(interface_path.stem)]
 include_base = '/'.join(include_parts)
 
+interface_type = "__".join(interface_path.parents[0].parts)
+interface_code = convert_camel_case_to_lower_case_underscore(interface_path.stem)
+
 header_files = [
-    'cstddef',
+    'stddef.h',
     'rosidl_runtime_c/message_type_support_struct.h',
     include_base + '__struct.h',
     include_base + '__type_support.h',
@@ -91,7 +94,7 @@ static _@(message.structure.namespaced_type.name)_type_support_data_t _@(message
 
 static const type_support_map_t _@(message.structure.namespaced_type.name)_message_typesupport_map = {
   @(len(type_supports)),
-  "@(package_name)",
+  "@(package_name)__@(interface_type)__@(interface_code)",
   &_@(message.structure.namespaced_type.name)_message_typesupport_ids.typesupport_identifier[0],
   &_@(message.structure.namespaced_type.name)_message_typesupport_symbol_names.symbol_name[0],
   &_@(message.structure.namespaced_type.name)_message_typesupport_data.data[0],